checks/apps/quantumespresso/quantumespresso_check.py

```diff
@@ -1,136 +1,212 @@
+# Copyright 2016-2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
 # ReFrame Project Developers. See the top-level LICENSE file for details.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
-import pathlib
-import re
-import sys
+import os
+import shutil
 
 import reframe as rfm
 import reframe.utility.sanity as sn
+import reframe.utility.udeps as udeps
+import uenv
 
-sys.path.append(str(pathlib.Path(__file__).parent.parent.parent / 'mixins'))
-from extra_launcher_options import ExtraLauncherOptionsMixin
-from sarus_extra_launcher_options import SarusExtraLauncherOptionsMixin
-from cuda_visible_devices_all import CudaVisibleDevicesAllMixin
-
-qe_tests = {
-    # R:total number of MPI ranks, T:threads per rank, P:walltime
-    'Au-surf': {
-        'zen2': {
-            # zen2 cpu nodes: 2sockets, 8 numa, 16c/numa = 128c (no MT)
-            'energy_reference': -11427.09017218,
-            'performance_reference': [{'R': 32, 'T': 4, 'P': 102.0}]
-        },
-        'zen3-4x-gpu-sm_80': {
-            # A100 nodes: 1socket, 4 numa, 16c/numa = 64c (no MT)
-            'energy_reference': -11427.09017218,
-            'performance_reference': [{'R': 4, 'T': 16, 'P': 32.0}]
+
+qe_references = {"Au surf": {"gh200": {"time_run": (20.40, None, 0.05, "s")}}}
+
+
+slurm_config = {
+    "Au surf": {
+        "gh200": {
+            "nodes": 1,
+            "ntasks-per-node": 4,
+            "cpus-per-task": 72,
+            "walltime": "0d0h20m0s",
+            "gpu": True,
         }
     }
 }
 
 
-class QuantumESPRESSOBase(rfm.RunOnlyRegressionTest):
-    energy_tolerance = 1.0e-6
-    executable = 'pw.x'
-    # TODO: tests should all have pw.in as input file
-    executable_opts = ['-in', 'ausurf.in']
+class qe_download(rfm.RunOnlyRegressionTest):
+    """
+    Download QE source code.
+    """
+
+    version = variable(str, value="7.3.1")
+    descr = "Fetch QE source code"
+    sourcedir = None
+    executable = "wget"
+    executable_opts = [
+        f"https://gitlab.com/QEF/q-e/-/archive/qe-{version}/q-e-qe-{version}.tar.gz",
+    ]
+    local = True
+
+    @sanity_function
+    def validate_download(self):
+        return sn.assert_eq(self.job.exitcode, 0)
 
-    @run_after('init')
-    def set_description(self):
-        self.descr = f'QuantumESPRESSO {self.test_name} check'
 
-    @run_before('run')
-    def set_parallel_resources(self):
+@rfm.simple_test
+class QeBuildTest(rfm.CompileOnlyRegressionTest):
+    """
+    Test QE build from source.
+    """
+
+    descr = "QuantumESPRESSO Build Test"
+    valid_prog_environs = ["+qe-dev"]
+    valid_systems = ["*"]
+    build_system = "CMake"
+    sourcesdir = None
+    maintainers = ["SSA"]
+    qe_sources = fixture(qe_download, scope="session")
+    build_locally = False
+    tags = {"uenv"}
+    pwx_executable = "./mps-wrapper.sh pw.x"
+    extra_resources = {"reservation": {"reservation": "NCCL"}}
+    time_limit = "0d0h30m0s"
```

Comment:
the `time_limit` here seems to have no effect, I had to change it in the global config for todi as workaround